kolibri/auth/permissions/auth.py

```diff
@@ -125,3 +127,59 @@ def readable_by_user_filter(self, user, queryset):
             return queryset.filter(id=user.dataset_id)
         else:
             return queryset.none()
+
+
+class CoachesCanManageGroupsForTheirClasses(BasePermissions):
+    def _user_is_coach_for_classroom(self, user, obj):
+        # make sure the target object is a group and user is a coach for the group's classroom
+        return obj.kind == LEARNERGROUP and user.has_role_for_collection(COACH, obj.parent)
+
+    def user_can_create_object(self, user, obj):
+        return self._user_is_coach_for_classroom(user, obj)
+
+    def user_can_read_object(self, user, obj):
+        return self._user_is_coach_for_classroom(user, obj)
+
+    def user_can_update_object(self, user, obj):
+        return self._user_is_coach_for_classroom(user, obj)
+
+    def user_can_delete_object(self, user, obj):
+        return self._user_is_coach_for_classroom(user, obj)
+
+    def readable_by_user_filter(self, user, queryset):
+        return HierarchyRelationsFilter(queryset).filter_by_hierarchy(
+            source_user=user,
+            role_kind=COACH,
+            descendant_collection=F("parent"),
+        ).filter(kind=LEARNERGROUP)
```

Comment:
This can just be `return queryset.none()`.

See comment https://github.com/learningequality/kolibri/pull/1143/files#r109480734